staging/src/k8s.io/apimachinery/pkg/util/net/http.go

```diff
@@ -387,13 +388,10 @@ redirectLoop:
 
 		resp.Body.Close() // not used
 
-		// Reset the connection.
-		intermediateConn.Close()
```

Comment:
Distributing this before every return increases the chance of future bugs...

Reply:
I realized this is captured by the defer statement on line 338

Reply:
It was largely copied from go's internal http client implementation. However, looking back at that it looks like they reorganized that code quite a bit. It's probably worth a follow up to to sync some of those improvements to here.